src/asset-swapper/utils/market_operation_utils/index.ts

```diff
@@ -138,6 +138,33 @@ export class MarketOperationUtils {
         return this.getMarketBuyLiquidity(makerToken, takerToken, limitOrders, amount, opts);
     }
 
+    /**
+     * Get the token amount per wei for native token. Note this function does not perform a full search (only perform
+     * a direct price check on `token` and native token) and calls `sampler.getBestNativeTokenSellRate` internally.
+     *
+     * @param token The token to check price for.
+     * @param opts GetMarketOrdersOpts object.
+     * @returns Token amount per wei.
+     */
+    public async getTokenAmountPerWei(token: string, opts?: Partial<GetMarketOrdersOpts>): Promise<BigNumber> {
```

Comment:
Two things:
- Functional: can we short circuit on Native Token and Wrapped Native token and just return 1? 
- Comment: so we sometimes call the output of this function `{input|output}AmountPerEth`, but here it's explicitly `PerWei` - can you leave a comment about the units?

Reply:
- should be able to short circuit on native / wrapped native token. There might an edge case where wrapped native token <-> native token is not close to 1 (however, i think extremely unlikely though)
- Yep I intentionally named the function & variables `perWei`. The naming for `{input|output}AmountPerEth` is confusing and it actually intends to mean `perWei` (this has caused some confusions when people read the code). I did not change the existing `AmountPerEth` name in this PR as I want to limit this PR to only metaTxn sell token fees. However, I think I will come up later and update the existing names to be `perWei`